client/txn.go

```diff
@@ -289,6 +279,25 @@ func (txn *Txn) Run(b *Batch) error {
 	return sendAndFill(txn.send, b)
 }
 
+func (txn *Txn) commit() error {
+	return txn.sendEndTxnCall(true /* commit */)
+}
+
+func (txn *Txn) cleanup(err error) {
+	if err != nil {
+		if replyErr := txn.Rollback(); replyErr != nil {
+			log.Errorf("failure aborting transaction: %s; abort caused by: %s", replyErr, err)
+		}
+	}
+}
+
+// CommitNoCleanup is the same as Commit but will not attempt to clean
+// up on failure. It is exposed only for use in txn_correctness_test.go
+// because those tests manipulate transaction state at a low level.
+func (txn *Txn) CommitNoCleanup() error {
```

Comment:
I'm not sure of the plan to "fix" the way `sql` uses `Txn`. I appreciate that to make progress we sometimes need to submit things in a staged manner, but that doesn't seem to be the case here. Felt more like the PR and the review were rushed through.